### PR TITLE
Fix/sa init

### PR DESCRIPTION
**Pull request by andre-merzky**
This branch has mostly been merged in a `hotfix`, but a small patch remained unmerged.  That introduces an `output` method on the component base class which, as the name suggests, allows a component to push something on an output queue.  That was always possible - the method just formalizes that.

### CODE REVIEW

**Review by iparask — CHANGES_REQUESTED** [review]
I think a small test for `output` will be welcome!

**Comment by andre-merzky** [comment]
> I think a small test for `output` will be welcome!

Do you have an suggestion on what that test would look like?  If you have the time, would you mind providing one?

**Comment by iparask** [comment]
> > I think a small test for `output` will be welcome!
> 
> Do you have a suggestion on what that test would look like? If you have the time, would you mind providing one?

I just pushed a small test. It misses a positive one, where something is put in the `_outputs` dictionary because I do not know the data type of its values. Does it help?

**Comment by andre-merzky** [comment]
> I just pushed a small test. It misses a positive one, where something is put in the `_outputs` dictionary because I do not know the data type of its values. Does it help?

Thanks, that works for me.  I added some test cases.

File: tests/test_components/test_component.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+        component._outputs = {'test_state':[]}
+
+        with self.assertRaises(ValueError):
+            component.output(things='test_thing')
+
+        with self.assertRaises(ValueError):
+            component.output(things='test_thing', state='test_state2')
+
+        component.output(things=None,  state='test_state')
```
Comment (iparask) [inline tests/test_components/test_component.py:42]: How do you expect `component._output` to look like after executing this line? After every call, an assert can be evaluated.

For example:
```
expected_output = ...
self.assertEqual(component._output, expected_output)
```

Comment (andre-merzky) [reply tests/test_components/test_component.py:42]: `_outputs` is a list of ZMQ endpoints.  The endpoints are stateless and don't change after the send.  We can't really check the entities unless we actually create ZMQ endpoints, so I think this is ok as is?

Comment (iparask) [reply tests/test_components/test_component.py:32]: It makes sense.

**Comment by iparask** [comment]
For some reason, all the tests fail after the cleanup. Let me fix that.

**Review by iparask — APPROVED** [review]
I think this PR is ready!

**Comment by andre-merzky** [comment]
What is the effect of enabling protected access (and others) at the end of the files?  I thought the scope for those instructions is limited by the file anyway?

**Comment by mturilli** [comment]
Can we merge this?

**Comment by andre-merzky** [comment]
> Can we merge this?

yes :-)

**Pull request merged**